src/HassClient/Helpers/Json/JsonExtensions.cs

```diff
@@ -132,27 +143,60 @@ HassService getServiceFields(string service, JsonElement element)
                         case "description":
                             serviceDescription = serviceProperty.Value.GetString();
                             break;
+                        case "required":
+                            required = serviceProperty.Value.GetBoolean();
+                            break;
                         case "fields":
                             foreach (var fieldsProperty in serviceProperty.Value.EnumerateObject())
                             {
                                 serviceFields.Add(getField(fieldsProperty.Name, fieldsProperty.Value));
                             }
-
+                            break;
+                        case "target":
+                            target = getSelector(serviceProperty.Name, serviceProperty.Value) as TargetSelector;
                             break;
                     }
                 }
                 return new HassService
                 {
                     Service = service,
                     Fields = serviceFields,
-                    Description = serviceDescription
+                    Description = serviceDescription,
+                    Required = required,
+                    Target = target
                 };
             }
 
+            object? getSelector(string selectorName, JsonElement element)
+            {
+                var selectorType = executingAssemblyTypes.FirstOrDefault(x => string.Equals($"{selectorName}Selector", x.Name, StringComparison.OrdinalIgnoreCase));
+
+                if (selectorType is null)
+                {
+                    return null;
+                }
+
+                if (element.ValueKind == JsonValueKind.Object && !element.EnumerateObject().Any() ||

```

Comment:
So If I understand correct
selector=  { Text = null}
Will result in The Text property != null

Reply:
yes